quic/server/handshake/ServerHandshake.cpp

```diff
@@ -350,132 +298,183 @@ void ServerHandshake::processPendingEvents() {
+        *server_.state_.keyScheduler(),
+        *server_.state_.cipher(),
+        folly::range(secretAvailable.secret.secret),
+        kQuicKeyLabel,
+        kQuicIVLabel);
+    auto headerCipher = server_.cryptoFactory_->makePacketNumberCipher(
+        folly::range(secretAvailable.secret.secret));
+    switch (secretAvailable.secret.type.type()) {
+      case fizz::SecretType::Type::EarlySecrets_E:
+        switch (*secretAvailable.secret.type.asEarlySecrets()) {
+          case fizz::EarlySecrets::ClientEarlyTraffic:
+            server_.zeroRttReadCipher_ = FizzAead::wrap(std::move(aead));
+            server_.zeroRttReadHeaderCipher_ = std::move(headerCipher);
+            break;
+          default:
+            break;
+        }
+        break;
+      case fizz::SecretType::Type::HandshakeSecrets_E:
+        switch (*secretAvailable.secret.type.asHandshakeSecrets()) {
+          case fizz::HandshakeSecrets::ClientHandshakeTraffic:
+            server_.handshakeReadCipher_ = FizzAead::wrap(std::move(aead));
+            server_.handshakeReadHeaderCipher_ = std::move(headerCipher);
+            break;
+          case fizz::HandshakeSecrets::ServerHandshakeTraffic:
+            server_.handshakeWriteCipher_ = FizzAead::wrap(std::move(aead));
+            server_.handshakeWriteHeaderCipher_ = std::move(headerCipher);
+            break;
+        }
+        break;
+      case fizz::SecretType::Type::AppTrafficSecrets_E:
+        switch (*secretAvailable.secret.type.asAppTrafficSecrets()) {
+          case fizz::AppTrafficSecrets::ClientAppTraffic:
+            server_.oneRttReadCipher_ = FizzAead::wrap(std::move(aead));
+            server_.oneRttReadHeaderCipher_ = std::move(headerCipher);
+            break;
+          case fizz::AppTrafficSecrets::ServerAppTraffic:
+            server_.oneRttWriteCipher_ = FizzAead::wrap(std::move(aead));
+            server_.oneRttWriteHeaderCipher_ = std::move(headerCipher);
+            break;
+        }
+        break;
+      case fizz::SecretType::Type::MasterSecrets_E:
+        break;
+    }
+    server_.handshakeEventAvailable_ = true;
+  }
 
-void ServerHandshake::ActionMoveVisitor::operator()(
-    fizz::server::MutateState& mutator) {
-  mutator(server_.state_);
-}
+ private:
+  ServerHandshake& server_;
+};
 
-void ServerHandshake::ActionMoveVisitor::operator()(
-    fizz::server::AttemptVersionFallback&) {
-  CHECK(false) << "Fallback Unexpected";
-}
+void ServerHandshake::processActions(
+    fizz::server::ServerStateMachine::CompletedActions actions) {
+  // This extra DestructorGuard is needed due to the gap between clearing
+  // actionGuard_ and potentially processing another action.
+  folly::DelayedDestruction::DestructorGuard dg(conn_);
 
-void ServerHandshake::ActionMoveVisitor::operator()(fizz::EndOfData&) {
-  server_.onError(std::make_pair(
-      "Unexpected close notify received", TransportErrorCode::INTERNAL_ERROR));
-}
+  ActionMoveVisitor visitor(*this);
```

Comment:
This is what was changed. The visitor used to be a field of the handshake and now it is a local variable. Everything else is the same.